app/src/androidTest/java/ch/epfl/sdp/musiconnect/MusicianFinderPageTests.java

```diff
@@ -0,0 +1,85 @@
+import androidx.test.espresso.intent.Intents;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.rule.ActivityTestRule;
+import androidx.test.uiautomator.UiDevice;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import ch.epfl.sdp.R;
+
+import static androidx.test.espresso.Espresso.closeSoftKeyboard;
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.clearText;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.scrollTo;
+import static androidx.test.espresso.action.ViewActions.typeText;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+
+/**
+ * @author Manuel Pellegrini, EPFL
+ */
+@RunWith(AndroidJUnit4.class)
+public class MusicianFinderPageTests {
+
+    private String firstName = "John";
+    private String lastName = "Lennon";
+    private String userName = "JohnLennon";
+
+    @Rule
+    public final ActivityTestRule<MusicianFinderPage> musicianFinderPageRule = new ActivityTestRule<>(MusicianFinderPage.class);
+
+    private UiDevice device;
+
+    // Before and after methods are used in order to accept tests with intents
+    @Before
+    public void initIntents() {
+        Intents.init();
+        device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+        MapsLocationTest.clickAlert(device);
+    }
+
+    @After
+    public void releaseIntents() { Intents.release(); }
+
+    @Test
+    public void testEditTextFieldsOfMusicianFinderWork() {
+        onView(withId(R.id.myMusicianFinderFirstNameID)).perform(ViewActions.scrollTo()).perform(clearText(), typeText(firstName));
+        onView(withId(R.id.myMusicianFinderLastNameID)).perform(ViewActions.scrollTo()).perform(clearText(), typeText(lastName));
+        onView(withId(R.id.myMusicianFinderUserNameID)).perform(ViewActions.scrollTo()).perform(clearText(), typeText(userName));
+
+        onView(withId(R.id.myMusicianFinderFirstNameID)).check(matches(withText(firstName)));
+        onView(withId(R.id.myMusicianFinderLastNameID)).check(matches(withText(lastName)));
+        onView(withId(R.id.myMusicianFinderUserNameID)).check(matches(withText(userName)));
+
+        closeSoftKeyboard();
+
+        onView(withId(R.id.myMusicianFinderFirstNameID)).check(matches(withText(firstName)));
+        onView(withId(R.id.myMusicianFinderLastNameID)).check(matches(withText(lastName)));
+        onView(withId(R.id.myMusicianFinderUserNameID)).check(matches(withText(userName)));
+    }
+
+    @Test
+    public void testSpinnerFieldsOfMusicianFinderWork() {
+        // TODO
+        assert(true);
```

Comment:
Why keeping this test?

Reply:
I wrote the name of the test to avoid to forget it, but I still have to write its body.